src/Schema/PostgreSQLSchemaManager.php

```diff
@@ -610,13 +610,49 @@ protected function selectTableNames(string $databaseName): Result
 
     protected function selectTableColumns(string $databaseName, ?string $tableName = null): Result
     {
+        // feature detection: if attrgenerated exists it means we are
+        // in a postgreSQL version supporting GENERATED columns
+        // see https://pgpedia.info/g/generated-column.html#implementation
+        $results                 = $this->_conn->executeQuery(<<<'SQL'
+            SELECT true
+            FROM information_schema.columns
+            WHERE table_name = 'pg_attribute'
+                AND column_name = 'attgenerated';
+        SQL);
+        $supportsGeneratedColumn = $results->fetchOne();
```

Comment:
We usually don't do this kind of feature detection. We use the platform class hierarchy to deal with functionality that is not available on all versions of a database.

Reply:
Ok no problem

So you prefer I   introduce a postgresqlplatfrom120 class ?

Reply:
Absolutely, yes!

Reply:
@derrabus  done !